monkey/monkey_island/cc/dal/IRepository.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABC
+from typing import Any, Optional, Sequence
+
+from monkey_island.cc.models import Config, Monkey
+from monkey_island.cc.models.edge import Edge
+
+
+class IRepository(ABC):
+
+    # Config
+    ###############################################
+
+    # This returns the current config
+    # TODO investigate if encryption should be here or where
+    def get_config(self) -> dict:
+        pass
+
+    def set_config(self, config: dict):
```

Comment:
Lets add docstring to every definition.

Reply:
Do we have a guideline about docstringing all interfaces?

Reply:
+1

Reply:
https://github.com/guardicore/monkey/issues/1952